parsing/parser.mly

```diff
@@ -2294,13 +2329,16 @@ simple_core_type2:
       { mktyp(Ptyp_variant(List.rev $3, Closed, Some [])) }
   | LBRACKETLESS opt_bar row_field_list GREATER name_tag_list RBRACKET
       { mktyp(Ptyp_variant(List.rev $3, Closed, Some (List.rev $5))) }
-  | LPAREN MODULE ext_attributes package_type RPAREN
-      { mktyp_attrs (Ptyp_package $4) $3 }
+  | LPAREN MODULE ext_attributes package_type_desc RPAREN
```

Comment:
I'm not fond of having `_desc` show up in a production, personally I think that it makes the production slightly less readable (`_desc` should only be internal wrapping). I would rather use `package_type` everywhere and `wrap_typ_attrs` in the semantic action here.

Reply:
I had missed that `wrap_typ_attrs`, thanks a lot!

Reply:
Actually, that change makes the locations slightly worse.
For:
```ocaml
type t =
  (module%foo[@foo] M)
```
I get this diff on the parsetree:
```diff
           Some
             core_type ([2,2]..[2,22]) ghost
               Ptyp_extension "foo"
-              core_type ([2,2]..[2,22])
+              core_type ([2,20]..[2,21])
                 attribute "foo"
                   []
                 Ptyp_package "M" ([2,20]..[2,21])
```

Reply:
I added a call to `reloc_typ` which fixed that issue.

Reply:
I tend to agree, though it is not really uniform in the rest of the parser.
And the other rules in `simple_core_type2` also call `mktyp` to make a type from a desc, so to me it looks like `_desc` might be the right construction.